### ISSUE TITLE
Is it possible to compile Unicorn Engine for a wasm target

### ISSUE THREAD

**Issue by rmccrystal** [unicorn-engine/unicorn#1709]
Would it be possible to use Unicorn Engine in a wasm environment, specifically with the Rust bindings? Compiling the Rust bindings with `--target wasm32-unknown-unknown` doesn't seem to work as gcc complains about not supporting the target.

**Comment by PhilippTakacs** [comment]
No because currently there is no wasm/js support for TCG. You might want to look at [unicorn.js](https://alexaltea.github.io/unicorn.js/) or at a qemu js/wasm fork.

**Comment by wtdcode** [comment]
Yes and no. The work is not that hard but most developers including me lacks enough time.
